### PR TITLE
Filter

### PR DESCRIPTION
**Pull request by yvlange**
- Add filter function
- Final touches on css

### CODE REVIEW

File: src/lib/api.js [hunk 1]
```diff
@@ -1,8 +1,22 @@
 export function fetchCharacters() {
-  const urlCharacters = "https://rickandmortyapi.com/api/character";
-  const main = document.querySelector("main");
+  let url;
+  const filterDropdown = document.querySelector(".header__filter");
+  const type = filterDropdown.value;
+  const searchBar = document.querySelector(".header__search-bar");
+  const name = searchBar.value;
+  console.log(name);
+  if (type === "alive") {
```
Comment (FriederikeM) [inline src/lib/api.js:8]: You can put this into its own function, that returns the correct url depending on the status of the characters. Then call it here

File: src/lib/api.js [hunk 2]
```diff
@@ -12,7 +26,9 @@ export function fetchCharacters() {
       data.results.forEach((characters) => {
         const section = document.createElement("section");
```
Comment (FriederikeM) [inline src/lib/api.js:27]: you can also put the creation of the character card in its own function (potentially in a new file -> import) and call it here

File: src/lib/api.js [hunk 3]
```diff
@@ -21,29 +37,40 @@ export function fetchCharacters() {
...
+
+        info.append(h2);
+        section.append(img);
+        section.append(info);
+        info.append(status);
+        info.append(species);
+        info.append(gender);
+
         if (characters.status === "Alive") {
           section.style.backgroundColor = "#cdfbbb";
```
Comment (FriederikeM) [inline src/lib/api.js:63]: maybe you can also use color variables here

File: src/styles/_base.css [hunk 4]
```diff
@@ -64,41 +72,66 @@ header {
...
+  background-repeat: no-repeat;
+  background-size: contain;
 }
 .characters {
   display: flex;
   width: 100vw;
   justify-content: space-between;
+  align-items: center;
+  padding: 5px;
   background-color: var(--background-color-characters);
```
Comment (FriederikeM) [inline src/styles/_base.css:91]: Not sure you need to define the background-color here if you are changing it dynamically in the js

**Review by FriederikeM — COMMENTED** [review]
Looks good :) I left a few comments, mainly passing on what Martin commented on mine :)

File: src/styles/_base.css (hunk 4 above)
Comment (yvlange) [reply src/styles/_base.css:91]: Thank you for your comments! :)

**Pull request merged**